packages/platform-core/src/usage-data/noop_usage_data_emitter.ts

```diff
@@ -0,0 +1,22 @@
+import { UsageDataEmitter } from './usage_data_emitter_factory';
+
+/**
+ * class implementing UsageDataEmitter that does not emit any data.
+ */
+export class NoOpUsageDataEmitter implements UsageDataEmitter {
+  /**
+   * no-op emitSuccess
+   */
+  emitSuccess(): Promise<void> {
+    // no-op
+    return Promise.resolve();
+  }
+
+  /**
+   * no-op emitFailure
+   */
+  // eslint-disable-next-line @typescript-eslint/no-unused-vars
+  emitFailure(_: Error): Promise<void> {
```

Comment:
```suggestion
  emitFailure(): Promise<void> {
```

one can skip parameters when overriding.